vm/asm.go

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2018 De-labtory
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package vm
+
+// Converts rawByteCode to assembly code.
+func assemble(rawByteCode []byte) (*asm, error) {
+	analysis()
```

Comment:
why don't you just return `asmReader` instead of `*ast`?

Reply:
The recommendation for golang is to get the interface as parameter and return the concrete implementation. This is more flexible way.

Reply:
ok! then.. LGTM!!